### PR TITLE
Change to CRA frontend

### PR DESCRIPTION
**Pull request by harshrajsoni05**
Converted project from VITE to CRA.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (PiyushK-Nucleusteq) [inline package.json:1]: remove unnecessary dependencies from this file.

File: src/components/loginform.jsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
+import '../styles/LoginForm.css';
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+// CSS
```
Comment (PiyushK-Nucleusteq) [inline src/components/loginform.jsx]: remove all commented code from all files.

File: src/components/action.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import '../styles/Action.css';
```
Comment (PiyushK-Nucleusteq) [inline src/components/action.jsx:1]: remove all commented code.

File: src/_test_/CategoryPage.test.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    fireEvent.click(screen.getByText('No'));
+    expect(screen.queryByText('Add Category')).not.toBeInTheDocument();
+  });
+
+  test('shows success toast on adding a category', async () => {
+    renderWithRouter(<CategoryPage />);
+
+    fireEvent.click(screen.getByRole('button', { name: /Add Category/i }));
+    // Fill form with valid data and submit
```
Comment (PiyushK-Nucleusteq) [inline src/_test_/CategoryPage.test.jsx]: remove commented code from all files.

File: src/pages/CategoryPage.jsx [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+  const handleSubmitModal = (data) => {
+    if (modalType === "add") {
+      handleAddCategory(data);
+    } else if (modalType === "edit") {
+      handleEditCategory(data);
+    } 
+  };
+
+  const columns = [
```
Comment (PiyushK-Nucleusteq) [inline src/pages/CategoryPage.jsx]: define  constants on top.

File: src/pages/IssuancePage.jsx [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+        user_id: currentData.user_id,
+        book_id: currentData.book_id,
+        issue_date: currentData.issue_date,
+        return_date: updatedIssuance.return_date,
+        status: updatedIssuance.status,
+        issuance_type: updatedIssuance.issuance_type,
+      };
+
+      const response = await updateIssuance(currentData.id, issuanceToUpdate);
+      console.log("Issuance to update:", issuanceToUpdate); // Add this line to debug
```
Comment (PiyushK-Nucleusteq) [inline src/pages/IssuancePage.jsx]: remove console.log

File: src/pages/IssuancePage.jsx [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+    setCurrentData({});
+  };
+
+  const handleSubmitModal = (data) => {
+    if (modalType === "edit") {
+      handleEditIssuance(data);
+      console.log('data->>>>>>>>>>>>>',data)
```
Comment (PiyushK-Nucleusteq) [inline src/pages/IssuancePage.jsx]: remove console.log

File: src/redux/store.jsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+
+import { createStore } from 'redux'
+
+import rootReducer from './rootReducer'
+
+const store = createStore(rootReducer, 
+    // composeWithDevTools()
```
Comment (PiyushK-Nucleusteq) [inline src/redux/store.jsx]: remove commented code

File: src/pages/UserPage.jsx [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+  useEffect(() => {
+    getUsers();
+  }, [currentPage, searchTerm]);
+
+  const getBooks = async () => {
+    try {
+      const data = await fetchAllBooks();
+      setBooks(data || []);
+    } catch (error) {
+      showFailureToast("Can't fetch Books")
```
Comment (PiyushK-Nucleusteq) [inline src/pages/UserPage.jsx:99]: don't use hard coded message, use message coming from the backend.

File: package.json (hunk 1 above)
Comment (harshrajsoni05) [reply package.json:1]: done sir

File: src/components/loginform.jsx (hunk 2 above)
Comment (harshrajsoni05) [reply src/components/loginform.jsx]: done.

File: src/components/action.jsx (hunk 3 above)
Comment (harshrajsoni05) [reply src/components/action.jsx:1]: done

File: src/_test_/CategoryPage.test.jsx (hunk 4 above)
Comment (harshrajsoni05) [reply src/_test_/CategoryPage.test.jsx]: done

File: src/pages/CategoryPage.jsx (hunk 5 above)
Comment (harshrajsoni05) [reply src/pages/CategoryPage.jsx]: done.

File: src/pages/IssuancePage.jsx (hunk 6 above)
Comment (harshrajsoni05) [reply src/pages/IssuancePage.jsx]: done.

File: src/pages/IssuancePage.jsx (hunk 7 above)
Comment (harshrajsoni05) [reply src/pages/IssuancePage.jsx]: removed

File: src/redux/store.jsx (hunk 8 above)
Comment (harshrajsoni05) [reply src/redux/store.jsx]: done.

File: src/pages/UserPage.jsx (hunk 9 above)
Comment (harshrajsoni05) [reply src/pages/UserPage.jsx:99]: yes.did it already in the new push

**Pull request merged**